core/controllers/domain_objects_validator.py

```diff
@@ -125,9 +125,14 @@ def validate_change_dict_for_blog_post(
     if 'title' in change_dict:
         blog_domain.BlogPost.require_valid_title(
             change_dict['title'], True)
+    else:
+        raise base.BaseHandler.InvalidInputException('No title provided')
     if 'thumbnail_filename' in change_dict:
         blog_domain.BlogPost.require_valid_thumbnail_filename(
             change_dict['thumbnail_filename'])
+    else:
```

Comment:
Thumbnail is optional in a draft blog post, only if the blog post is published it is required.